File: NQT/Numbers/PrimeNumber/TwinPrime.java
package Numbers.PrimeNumber;

import java.util.Scanner;

public class TwinPrime {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the lower bound of the range: ");
        int lowerBound = scanner.nextInt();
        System.out.print("Enter the upper bound of the range: ");
        int upperBound = scanner.nextInt();
        System.out.println("Twin prime pairs between " + lowerBound + " and " + upperBound + ":");
        for (int number = lowerBound; number <= upperBound - 2; number++) {
            if (isPrime(number) && isPrime(number + 2)) {
                System.out.println("(" + number + ", " + (number + 2) + ")");
            }
        }
        scanner.close();
    }

    public static boolean isPrime(int n) {
        if (n <= 1) {
            return false; // Prime numbers are greater than 1
        }
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) {
                return false; // If divisible by any number other than 1 and itself, it's not prime
            }
        }
        return true; // If no divisors found, it's a prime number
    }
}

File: NQT/Numbers/PrimeNumber/twin_prime.py
# find the twin prime numbers in the given range
def is_prime(num):
    if num <= 1:
        return False
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            return False
    return True

def twin_primes_in_range(start, end):
    for num in range(start, end + 1):
        if is_prime(num) and is_prime(num + 2):
            print(f"({num}, {num + 2})")

def main():
    start = int(input("Enter the start of the range: "))
    end = int(input("Enter the end of the range: "))
    print(f"Twin prime numbers between {start} and {end} are:")
    twin_primes_in_range(start, end)

if __name__ == "__main__":
    main()